connect/runtime/src/main/java/org/apache/kafka/connect/runtime/RestartPlan.java

```diff
@@ -0,0 +1,154 @@
+     */
+    public RestartRequest restartRequest() {
+        return request;
+    }
+
+    /**
+     * Get the {@link ConnectorStateInfo} that reflects the current state of the connector <em>except</em> with the {@code status}
+     * set to {@link AbstractStatus.State#RESTARTING} for the {@link Connector} instance and any {@link Task} instances that
+     * are to be restarted, based upon the {@link #restartRequest() restart request}.
+     *
+     * @return the connector state info that reflects the restart plan; never null
+     */
+    public ConnectorStateInfo restartConnectorStateInfo() {
+        return stateInfo;
+    }
+
+    /**
+     * Get the immutable collection of {@link ConnectorTaskId} for all tasks to be restarted
+     * based upon the {@link #restartRequest() restart request}.
+     *
+     * @return the IDs of the tasks to be restarted; never null but possibly empty
+     */
+    public Collection<ConnectorTaskId> taskIdsToRestart() {
+        return idsToRestart;
+    }
+
+    /**
+     * Determine whether the {@link Connector} instance is to be restarted
+     * based upon the {@link #restartRequest() restart request}.
+     *
+     * @return true if the {@link Connector} instance is to be restarted, or false otherwise
+     */
+    public boolean shouldRestartConnector() {
+        return isRestarting(stateInfo.connector());
+    }
+
+    /**
+     * Determine whether at least one {@link Task} instance is to be restarted
+     * based upon the {@link #restartRequest() restart request}.
+     *
+     * @return true if any {@link Task} instances are to be restarted, or false if none are to be restarted
+     */
+    public boolean shouldRestartTasks() {
+        return !taskIdsToRestart().isEmpty();
+    }
+
+    /**
+     * Get the number of connector tasks that are to be restarted
+     * based upon the {@link #restartRequest() restart request}.
+     *
+     * @return the number of {@link Task} instance is to be restarted
+     */
+    public int restartTaskCount() {
+        return taskIdsToRestart().size();
+    }
+
+    /**
+     * Get the total number of tasks in the connector.
+     *
+     * @return the total number of tasks
+     */
+    public int totalTaskCount() {
+        return stateInfo.tasks().size();
+    }
+
+    private boolean isRestarting(ConnectorStateInfo.AbstractState state) {
+        return isRestarting(state.state());
+    }
+
+    private boolean isRestarting(String state) {
+        return AbstractStatus.State.RESTARTING.toString().equalsIgnoreCase(state);
+    }
+
+    @Override
+    public String toString() {
+        if (shouldRestartConnector()) {
+            return String.format(
+                    "plan to restart connector and %d of %d tasks for %s", restartTaskCount(), totalTaskCount(), request
+            );
+        }
+        return String.format(
+                "plan to restart %d of %d tasks for %s", restartTaskCount(), totalTaskCount(), request
+        );
+    }
```

Comment:
something like? 
```suggestion
return shouldRestartConnector()
               ? String.format("plan to restart connector and %d of %d tasks for %s", restartTaskCount(), totalTaskCount(), request)
               : String.format("plan to restart %d of %d tasks for %s", restartTaskCount(), totalTaskCount(), request);
```
breaking this long statement doesn't make it much less long I guess.